src/native/corehost/hostmisc/utils.cpp

```diff
@@ -350,6 +350,56 @@ bool try_stou(const pal::string_t& str, unsigned* num)
     return true;
 }
 
+#if defined(_WIN32)
+bool is_running_in_wow64()
+{
+    BOOL fWow64Process = FALSE;
+    if (!IsWow64Process(GetCurrentProcess(), &fWow64Process))
+    {
+        return false;
+    }
+    return (fWow64Process != FALSE);
+}
+
+typedef BOOL (WINAPI* is_wow64_process2)(
+  HANDLE hProcess,
+  USHORT *pProcessMachine,
+  USHORT *pNativeMachine
+);
+
+bool is_emulating_x64()
+{
+    USHORT pProcessMachine, pNativeMachine;
+    auto kernel32 = LoadLibraryEx("kernel32.dll", NULL, 0);
```

Comment:
```suggestion
    auto kernel32 = LoadLibraryExW(L"kernel32.dll", NULL, LOAD_LIBRARY_SEARCH_SYSTEM32);
```
- Unicode version (W suffix) is faster - avoids ANSI->Unicode conversion
- Use `LOAD_LIBRARY_SEARCH_SYSTEM32` to avoid static analysis warnings about dll hijacking